mmv1/third_party/terraform/services/vertexai/resource_vertex_ai_feature_group_test.go

```diff
@@ -0,0 +1,150 @@
+package vertexai_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+
+func TestAccVertexAIFeatureGroup_updated(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckVertexAIFeatureGroupDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVertexAIFeatureGroup_basic(context),
+			},
+			{
+				ResourceName:            "google_vertex_ai_feature_group.feature_group",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"etag", "region", "description", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccVertexAIFeatureGroup_updated(context),
+			},
+			{
+				ResourceName:            "google_vertex_ai_feature_group.feature_group",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"etag", "region", "description", "labels", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccVertexAIFeatureGroup_basic(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+  resource "google_vertex_ai_feature_group" "feature_group" {
+  name = "sample_feature_group_movie_reviews"
+  description = "A sample feature group"
+  region = "us-central1"
+  labels = {
+      label-one = "value-one"
+  }
+  big_query {
+    big_query_source {
+        # The source table must have a column named 'feature_timestamp' of type TIMESTAMP.
+        input_uri = "bq://${google_bigquery_table.sample_table.project}.${google_bigquery_table.sample_table.dataset_id}.${google_bigquery_table.sample_table.table_id}"
+    }
+    entity_id_columns = ["movie_id"]
+  }
+}
+
+resource "google_bigquery_dataset" "sample_dataset" {
+  dataset_id                  = "tf_test_job_load%{random_suffix}_dataset"
+  friendly_name               = "test"
+  description                 = "This is a test description"
+  location                    = "US"
+}
+
+resource "google_bigquery_table" "sample_table" {
+  deletion_protection = false
+  dataset_id = google_bigquery_dataset.sample_dataset.dataset_id
+  table_id   = "tf_test_job_load%{random_suffix}_table"
+
+  schema = <<EOF
+[
+    {
+        "name": "movie_id",
```

Comment:
also needs genericizing, and is the review column needed for feature group to function? It could be removed from all the samples if not